app/controllers/sessions_controller.rb

```diff
@@ -9,6 +11,7 @@ def index
 
   def show
     @session = sessions.friendly.find(params[:id])
+    raise ActiveRecord::RecordNotFound if @session.private? && !user_signed_in?
```

Comment:
another option would be this:
```suggestion
    if user_signed_in?
      @session = sessions.friendly.find(params[:id])
    else
      @session = sessions.publics.friendly.find(params[:id])
    end
```